[message] remove the AllMessageQueue in MessagePool

abtink opened the pull request:
This commit removes the all-messages queue from `MessagePool`. This
simplifies the `Message` header `MessageInfo` and allow a message
to be inserted into one linked list.

--------
The idea behind the all-message queue in `MessagePool` was for us to have a way to track all allocated messages and can possibly evict/drop a low-priority message to make room for other higher priority ones. Currently we allow eviction from the `mSendQueue` priority queue in `MeshForwarder` but the all-message queue is not really used for this. Talking with @jwhui , we decided to remove this feature from code. When needed we can get to all messages by keeping track of a list of message queues from which we allow message eviction. This model let us be selective and permit eviction from certain queues and disallow this behavior on other message queues.

jwhui commented on src/core/common/message.hpp:
```diff
@@ -1020,30 +1008,27 @@ class MessageQueue : public otMessageQueue
      * This method adds a message to the end of the list.
      *
-     * @param[in]  aListId   The list to add @p aMessage to.
      * @param[in]  aMessage  The message to add to @p aListId.
```
Remove `@p aListId` reference.

jwhui commented on src/core/common/message.hpp:
```diff
@@ -1020,30 +1008,27 @@ class MessageQueue : public otMessageQueue
      * This method adds a message at a give position (head or tail) of the list.
      *
-     * @param[in]  aListId   The list to add @p aMessage to.
      * @param[in]  aMessage  The message to add to @p aListId.
```
Remove `@p aListId` reference.

jwhui commented on src/core/common/message.hpp:
```diff
@@ -1020,30 +1008,27 @@ class MessageQueue : public otMessageQueue
      * This method removes a message from a list.
      *
-     * @param[in]  aListId   The list to add @p aMessage to.
      * @param[in]  aMessage  The message to add to @p aListId.
```
Remove `@p aListId` reference.

jwhui commented on src/core/common/message.hpp:
```diff
@@ -1125,20 +1110,18 @@ class PriorityQueue
      * This method adds a message to a list.
      *
-     * @param[in]  aListId   The list to add @p aMessage to.
      * @param[in]  aMessage  The message to add to @p aListId.
```
Remove `@p aListId` reference.

jwhui commented on src/core/common/message.hpp:
```diff
@@ -1125,20 +1110,18 @@ class PriorityQueue
      * This method removes a message from a list.
      *
-     * @param[in]  aListId   The list to add @p aMessage to.
      * @param[in]  aMessage  The message to add to @p aListId.
```
Remove `@p aListId` reference.

abtink commented:
@jwhui , updated the docs. Also added a second commit on this PR which inlines the `AddToList` and `RemoveFromList` methods into `Enqueue` and `Dequeue` methods (and removes them). 
(kept it seperate to make it easier to read/review). Can you please squash it into the other commit before merge? Thanks.
